--- casper	2015-09-15 05:04:34.000000000 -0700
+++ casper-encsqfs	2015-10-25 21:33:42.000000000 -0700
@@ -494,7 +494,8 @@
                 # it is a plain directory: do nothing
                 rofsstring="${image}=${roopt}:${rofsstring}"
                 rofslist="${image} ${rofslist}"
-            elif [ -f "${image}" ]; then
+
+            elif [ -e "${image}" ]; then # ENCSQFS changed from -f
                 backdev=$(get_backing_device "$image")
                 fstype=$(get_fstype "${backdev}")
                 if [ "${fstype}" = "unknown" ]; then
@@ -732,6 +733,18 @@
         done
     fi
 
+    # ENCSQFS
+    # Symlink was found and thus /cdrom was mounted, now set up crypto
+    # to get /dev/mapper/squashfs so symlink is valid
+    maybe_break presquashfs
+    if [ -f /cdrom/casper/filesystem.squashfs.luks ] ; then
+	/bin/plymouth ask-for-password --prompt Password | \
+            /sbin/cryptsetup -T 1 --key-file=- luksOpen \
+	                     /cdrom/casper/filesystem.squashfs.luks \
+                             squashfs
+    fi
+    maybe_break squashfs
+
     if [ -z "${livefs_root}" ]; then
         panic "Unable to find a medium containing a live file system"
     fi
